src/statemachine.rs

```diff
@@ -80,50 +70,37 @@ impl TestState {
     }
 
     /// Updates the state machine state
-    pub fn step<T:StateData>(self, data: &mut T, config: &Config) -> TestState {
+    pub fn step<T:StateData>(self, data: &mut T, config: &Config) -> Result<TestState, RunError> {
         match self {
             TestState::Start{start_time} => {
-                if let Some(s) = data.start() {
-                    s
-                } else if start_time.elapsed() >= config.test_timeout {
-                    println!("Error: Timed out when starting test");
-                    TestState::Timeout
+                data.start()?;
```

Comment:
You need to save this returned state somewhere otherwise the `TestState::Initialise` won't be returned